uvicorn/main.py

```diff
@@ -403,11 +400,13 @@ def main(
         "headers": [header.split(":", 1) for header in headers],
         "use_colors": use_colors,
         "factory": factory,
+        "app_dir": app_dir,
     }
     run(app, **kwargs)
 
 
 def run(app: typing.Union[ASGIApplication, str], **kwargs: typing.Any) -> None:
+    sys.path.insert(0, kwargs.pop("app_dir", "."))
```

Comment:
Nop, we want to change the priority between "current directory" and `app_dir`.

Reply:
@Kludex ok, but if app_dir is not specified at all (i.e. you default to "."), should you be changing the priority in that case? I don't think so...

Reply:
I'm not sure, but then we'd want something like this:

```python
    app_dir = kwargs.pop("app_dir")
    if app_dir is not None:
        sys.path.insert(0, app_dir)
```

This PR doesn't aim to change that logic tho, the aim here is to match the behavior from the CLI to the `run()` function, so I guess that would be a separate discussion.

Reply:
@Kludex your suggested change in the above comment is the least that is needed, and it is _not_ changing the logic. The `main` function still gives a default value of `.`, so cli logic is no different than it was before. No need to add _another_ default of `.` in the `run` function as well (because if you do, then the run function would have backwards-incompatible behavior, as well as giving devs no way to opt-out of modifying sys.path... and _worse_: no way to opt-out of modifying sys.path[0] [which is not a good idea in the first place](https://stackoverflow.com/a/10097543/2599133)).

Reply:
Yeah, you're right. Thanks. I'll apply the changes.